pkg/build/builder/dockerutil.go

```diff
@@ -89,6 +120,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	return err
 }
 
+func getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error {
+	glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.")
+	timeout := time.After(time.Second * 2)
+
+waitLoop:
+	for {
```

Comment:
Ahh, I see, that would be a good fit. I turned it into wait.Until()
instead.

https://github.com/openshift/origin/pull/8195/files#diff-1af9f534a92f3f353b73f0ed0c3a0a50R149

It doesn't improve the situation at all. I think I'll revert to the
original for loop with some small improvements kept.

On 03/30/2016 01:42 PM, David Eads wrote:

> In pkg/build/builder/dockerutil.go <https://github.com/openshift/origin/pull/8195#discussion_r57874612>:
>
>> @@ -89,6 +120,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
>>  	return err
>>  }
>>
>> +func getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error {
>> +	glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.")
>> +	timeout := time.After(time.Second * 2)
>> +
>> +waitLoop:
>> +	for {
>
> This one: https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/debug.go#L280, but its a bad dependency to
> make here.
> … <#>
> On Wed, Mar 30, 2016 at 3:55 AM, Michal Minar <notifications@github.com> wrote: In pkg/build/builder/dockerutil.go
> <#8195 (comment) <https://github.com/openshift/origin/pull/8195#discussion_r57848937>>: > @@ -89,6 +120,35 @@ func
> pushImage(client DockerClient, name string, authConfig docker.AuthConfigura > return err > } > > +func
> getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error { > +
> glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.") > + timeout :=
> time.After(time.Second * 2) > + > +waitLoop: > + for { I believe you mean this Until
> <https://github.com/kubernetes/kubernetes/blob/master/pkg/util/wait/wait.go#L46>. I can update it at once. — You are
> receiving this because you commented. Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/009f353ce0f5825f8e3821be026ae9f687ab66dc#r57848937>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/009f353ce0f5825f8e3821be026ae9f687ab66dc#r57874612>
>